src/Components/StepThree/StepThree.js

```diff
@@ -9,75 +9,86 @@ import { agricultureSkillsInfo, agricultureSkillsWhatInfo, backButton, bicycleIn
 
 const StepThree = ({prevStep, setData, data, submitForm, step, getRealDate}) => {
 
-    const [inputs, setInputs] = useState({
-      doesHaveAgricultureSkills: data.doesHaveAgricultureSkills,
-      agricultureSkills: data.agricultureSkills,
-      doesHaveMetalworkSkills: data.doesHaveMetalworkSkills,
-      metalworkSkills: [],
-      isConvicted: data.isConvicted,
-      convictions: [
-        {
-         forWhat: "",
-         convictionDate: "",
-        }
-      ],
-      doesFlyAirplane: data.doesFlyAirplane,
-      doesDriveCar: data.doesDriveCar,
-      doesDriveBicycle: data.doesDriveBicycle,
-    });
+    const reducerStepThree = (state, action) => {
+        switch (action.type) {
+            case "agricultureSkills":
+                return {
+                    ...state,
+                    [action.type]: action.payload.target.value,
+                };
+            case "metalworkSkills":
+            case "convictions":
+                return {
+                    ...state,
+                    [action.type]: action.payload,
+                  };
+            case "doesHaveAgricultureSkills":
+            case "doesHaveMetalworkSkills":
+            case "isConvicted":
+            case "doesFlyAirplane":
+            case "doesDriveCar":
+            case "doesDriveBicycle":
+                return {
+                    ...state,
+                    [action.type]: !!action.payload.target.value,
+                  };
+            case "forWhat":
+            case "convictionDate":
+                return {
+                    ...state,
+                      convictions: {
+                            ...state.convictions,
+                            [action.type]: action.payload,
+                      }
+                    };
+            default:
+                throw new Error("Try again.");
+                    }
+                  };
 
-    const metalWorkSkillsTypes = ["marking", "cutting", "drilling", "cutThreads", "filling" ,"joining"];
+    const [state, dispatch] = useReducer(reducerStepThree, data);
```

Comment:
I would suggest using more descriptive name for state here - maybe something related to skills, or inputs? Just to indicate that this is an object that is storing the values for this page's inputs. Since the Context values are named "data" (that could also be a bit more descriptive too :) )